diff --git a/arch/arm64/boot/dts/renesas/r9a09g056.dtsi b/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
@@ -3300,6 +3300,19 @@
 			renesas,pcie-sys = <&sys>;
 			status = "disabled";
 		};
+
+		drpai0: drpai@16800000 {
+			compatible = "renesas,rzv2n-drpai";
+			reg = <0 0x17000000 0 0x1000000>,
+				<0 0x16800000 0 0x400000>;
+			interrupts = <GIC_SPI 914 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 915 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 916 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 917 IRQ_TYPE_LEVEL_HIGH>;
+			//resets = <&cpg R9A09G056_DRPAI_ARESETN>;
+			resets = <&cpg 253>;
+			status = "disabled";
+		};
 	};

 	thermal-zones {
diff --git a/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts b/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
@@ -1504,3 +1504,9 @@
 	status = "okay";
 };
 #endif
+
+&drpai0 {
+	memory-region = <&drp_reserved>;
+	memory-shared-for-drpai-ext-cont = <&shared_drp_reserved>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi
@@ -637,3 +637,9 @@
 	uart-has-rtscts;
 	status = "disabled";
 };
+
+&drpai0 {
+	memory-region = <&drp_reserved>;
+	memory-shared-for-drpai-ext-cont = <&shared_drp_reserved>;
+	status = "okay";
+};
